lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -153,10 +154,16 @@ defmodule Mix.Compilers.Elixir do
       |> List.delete(module)
       |> Enum.reject(&match?("elixir_" <> _, Atom.to_string(&1)))
 
+    runtime_remote_calls =
+      runtime_remote_calls
+      |> Enum.reject(&match?("elixir_" <> _, Atom.to_string(elem(&1, 0))))
```

Comment:
Maybe we shouldn't filter these here, actually? I guess a user could refer to an `:elixir_*` module in their own code improperly.